### PR TITLE
Roman penza patch 1

### PR DESCRIPTION
**Pull request by RomanPenza**
Added datetime_utils module that contains only one function at the moment: add_time function. This function takes a date as a string input in the 'yyyy-mm-dd' format and returns transpose of the date, given the specified number of days, months and years. To substract days, months, years from the date passed, use negative values.

### CODE REVIEW

File: codonPython/datetime_utils.py [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+     
+    >>>add_time('2018-03-04', days = -90)
+    '2017-12-04'
+
+    >>>add_time('2018-03-04', years = 2)
+    '2020-03-04'
+
+    """
+    
+    return str((datetime.datetime.strptime(date,'%Y-%m-%d') + relativedelta(days = int(days), months = int(months), years = int(years))).strftime(('%Y-%m-%d')))
```
Comment (georgeRobertson) [inline codonPython/datetime_utils.py:45]: Hi @RomanPenza,

Please could you ensure that there are no spaces around "=" when indicating a key argument. This is in compliance with PEP 8 formatting.

For example,

`relativedelta(days=int(days), months=int(months), years=int(years))`

**Review by georgeRobertson — CHANGES_REQUESTED** [review]
Hi @RomanPenza ,

Please could you make the changes requested in the review. Also, can you ensure that the build passes. Otherwise, I won't be able to merge this pull request.

Build error...

> ________________ ERROR collecting codonPython/datetime_utils.py ________________
> /opt/python/3.6.7/lib/python3.6/doctest.py:933: in find
>     self._find(tests, obj, name, module, source_lines, globs, {})
> ../../../virtualenv/python3.6.7/lib/python3.6/site-packages/_pytest/doctest.py:403: in _find
>     self, tests, obj, name, module, source_lines, globs, seen
> /opt/python/3.6.7/lib/python3.6/doctest.py:996: in _find
>     globs, seen)
> ../../../virtualenv/python3.6.7/lib/python3.6/site-packages/_pytest/doctest.py:403: in _find
>     self, tests, obj, name, module, source_lines, globs, seen
> /opt/python/3.6.7/lib/python3.6/doctest.py:983: in _find
>     test = self._get_test(obj, name, module, globs, source_lines)
> /opt/python/3.6.7/lib/python3.6/doctest.py:1067: in _get_test
>     filename, lineno)
> /opt/python/3.6.7/lib/python3.6/doctest.py:669: in get_doctest
>     return DocTest(self.get_examples(string, name), globs,
> /opt/python/3.6.7/lib/python3.6/doctest.py:683: in get_examples
>     return [x for x in self.parse(string, name)
> /opt/python/3.6.7/lib/python3.6/doctest.py:645: in parse
>     self._parse_example(m, name, lineno)
> /opt/python/3.6.7/lib/python3.6/doctest.py:703: in _parse_example
>     self._check_prompt_blank(source_lines, indent, name, lineno)
> /opt/python/3.6.7/lib/python3.6/doctest.py:790: in _check_prompt_blank
>     line[indent:indent+3], line))
> E   ValueError: line 29 of the docstring for codonPython.datetime_utils.add_time lacks blank after >>>: ">>>add_time('2018-03-04', months = 4)"

Thanks,
George

File: codonPython/datetime_utils.py (hunk 1 above)
Comment (RomanPenza) [reply codonPython/datetime_utils.py]: Hi, 

I get an error:
"Plugin 'pytest_cov' could not be loaded: (pytest 4.3.1 (/home/travis/virtualenv/python3.6.7/lib/python3.6/site-packages)"

Not sure where it comes from...

Comment (georgeRobertson) [reply codonPython/datetime_utils.py]: Hi @RomanPenza, Apologies. There is currently a bug within the master build regarding the docs deployment. We are aiming to fix it in the near future, but there are no given timescales at the moment as the majority of the team is busy. Once the bug has been resolved I will let you know and we can proceed.

Comment (georgeRobertson) [reply codonPython/datetime_utils.py]: Hi @RomanPenza 

We have fixed the bug with docs deployment. However, we have also updated our [contribution guide](https://github.com/codonlibrary/codonPython/blob/master/CONTRIBUTING.md) and your merge request currently conflicts with bullet point 3:
`Duplicated functionality. For example, if your function is already done by another well known package.`

In my opinion, the functionality that you have wrote can already be accessed using the datetime package, which is why I have challenged your merge request. If you disagree, please let me know by the 20th November or I will have to reject the merge request.

Apologies about this, I did originally agree to accept the function, but all active and future merge requests must adhere to the updated contribution guide.

**Comment by georgeRobertson** [comment]
Hi @RomanPenza 

We have fixed the bug with docs deployment. However, we have also updated our contribution guide and your merge request currently conflicts with bullet point 3:
`Duplicated functionality. For example, if your function is already done by another well known package.`

In my opinion, the functionality that you have wrote can already be accessed using the datetime package, which is why I have challenged your merge request. If you disagree, please let me know by the **20th November** or I will have to reject the merge request.

Apologies about this, I did originally agree to accept the function, but all active and future merge requests must adhere to the updated contribution guide.

**Pull request closed without merging**